checkstyle-checks/src/main/resources/sonatype/checkstyle-configuration.xml

```diff
@@ -25,8 +25,8 @@
 
   <module name="LineLength">
     <property name="max" value="120"/>
-    <!-- Allow longer lines in comments. -->
-    <property name="ignorePattern" value="^ \*.*"/>
+    <!-- Allow longer lines in block comments and whole string literals -->
+    <property name="ignorePattern" value='^ +\*|"[^"]{100,}"'/>
```

Comment:
fix: allow 1 or more space in block comments, so it works for block comment sat any level.
new: if the line is mostly a string of more than 100 characters, then leave it. these are often test inputs.

Reply:
examples that are allowed now for line length:
```
  /**
   * javadoc comments for fields and methods, not just class-level
   */
```
strings without unnecessary breaks:
```
    String id =
        "pkg:hf-model/org/repo/path%2Fto%2Fmodel@commit-hash?classifier=format&sonatype_repository=huggingface.co&type=ext";
```
parameters in CSV format in data-driven tests:
```
  @CsvSource({
      "conan, cctz, 2.3, , , pkg:conan/cctz@2.3?sonatype_naming_authority=conan",
      "bincrafters, cctz, 2.3, bincrafters, stable, pkg:conan/bincrafters/cctz@2.3?channel=stable&sonatype_naming_authority=bincrafters"
  })
```
if it's a shorter string in a long parameter list or something, that could be wrapped sensibly, then it's still triggered.